File: ProjetCompain/src/main/java/tn/iit/entites/Compain.java
package tn.iit.entites;

import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Compain {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String Reference;
	private String name;
	private String country;
	private String town;
	private String adress;
	private String pc;
	private Date createdAt;  
	private Date UpdatedAt;
}                                                   

File: compainNode/models/compain.js
const mongoose=require('mongoose');

    
    const compainSchema = new mongoose.Schema({
        reference: { type: String, required: true },
        name: { type: String, required: true },
        country: { type: String, required: true },
        town: { type: String, required: true },
        adress: { type: String, required: true },
        pc: { type: String, required: true },
      }, {
        timestamps: { createdAt: 'createdAt', updatedAt: 'updatedAt' } 
      });

      const compain=mongoose.model('compain',compainSchema);

      module.exports=compain;
